### PR TITLE
feat: add XPath support to getAnchor()

### PR DESCRIPTION
**Pull request by sir-kokabi**
#809

### CODE REVIEW

File: packages/wxt/src/client/content-scripts/ui/index.ts [hunk 1]
```diff
@@ -199,8 +199,25 @@ function getAnchor(options: ContentScriptAnchoredOptions): Element | undefined {
 
   let resolved =
     typeof options.anchor === 'function' ? options.anchor() : options.anchor;
-  if (typeof resolved === 'string')
-    return document.querySelector<Element>(resolved) ?? undefined;
+
+  if (typeof resolved === 'string') {
+    // If the string is an XPath expression (starts with '//' or '/')
+    if (resolved.startsWith('//') || resolved.startsWith('/')) {
```
Comment (aklinker1) [inline packages/wxt/src/client/content-scripts/ui/index.ts:205]: can this be simplified?

```suggestion
    if (resolved.startsWith('/')) {
```

File: packages/wxt/src/client/content-scripts/ui/types.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ export type ContentScriptPositioningOptions =
 
 export interface ContentScriptAnchoredOptions {
   /**
-   * A CSS selector, element, or function that returns one of the two. Along with `append`, the
+   * A CSS selector, XPath expression, element, or function that returns one of the two. Along with `append`, the
```
Comment (aklinker1) [inline packages/wxt/src/client/content-scripts/ui/types.ts:197]:
```suggestion
   * A CSS selector, XPath expression, element, or function that returns one of the three. Along with `append`, the
```

File: packages/wxt/src/client/content-scripts/ui/index.ts (hunk 1 above)
Comment (sir-kokabi) [reply packages/wxt/src/client/content-scripts/ui/index.ts]: Yes, you are correct. It has been resolved.

File: packages/wxt/src/client/content-scripts/ui/types.ts (hunk 2 above)
Comment (sir-kokabi) [reply packages/wxt/src/client/content-scripts/ui/types.ts]: This one too

**Review by aklinker1 — APPROVED** [review]

**Pull request merged**